azavea/pfb-network-connectivity
# Feature/awf/django save neighborhood geom

## Overview

Bah. This took longer than expected. Due to the abstractions in the Django file API, we can only rely on a very specific subset of File operations when working with a FileField. This means that in order to protect against the underlying storage API changing, we need to read the entire file to disk so we can parse it with Fiona or even something like Django's built-in DataSource API, because those only support Python file objects or a string path to a file on disk.

## Notes

Added a data migration to parse any existing Neighborhoods, catching all errors and printing exceptions to console so that migrations don't fail if there's an error loading up a Neighborhood's shapefile.

## Testing Instructions

1. Run migrations. All of your existing neighborhoods should now have geoms attached.

2. Create a new neighborhood boundary via the API, then inspect it in the django shell, `neighborhood.geom` should be non-null.

Closes #190
